python/main.py

```diff
@@ -235,3 +235,110 @@ def search_items(keyword: str = Query(..., min_length=1), db: sqlite3.Connection
     except Exception as e:
         logger.error(f"Failed to search items: {e}")
         raise HTTPException(status_code=500, detail="Failed to search items")
+
+
+@app.delete("/items/{item_id}")
+def delete_item(item_id: int, db: sqlite3.Connection = Depends(get_db)):
+    try:
+        cursor = db.cursor()
+        cursor.execute("SELECT image_name FROM items WHERE id = ?", (item_id,))
+        item = cursor.fetchone()
+
+        if item is None:
+            raise HTTPException(status_code=404, detail="Item not found")
+
+        image_name = item["image_name"]
+        image_path = images / image_name
+
+        cursor.execute("DELETE FROM items WHERE id = ?", (item_id,))
+        db.commit()
+
+        cursor.execute("SELECT COUNT(*) FROM items WHERE image_name = ?", (image_name,))
+        count = cursor.fetchone()[0]
+        
+        if count == 0 and image_path.exists():
+            os.remove(image_path)
+
+        return {"message": f"Item {item_id} deleted successfully"}
+
+    except Exception as e:
+        db.rollback()
+        logger.error(f"Failed to delete item {item_id}: {e}")
+        raise HTTPException(status_code=500, detail="Failed to delete item")
+
+
+@app.patch("/items/{item_id}")
```

Comment:
In patch, we can also check if the new value is equal to old value. If yes, we don't need to do a database query